app/services/personality_service.rb

```diff
@@ -3,9 +3,12 @@ def initialize(user)
         @user = user
     end
 
-    def buy(personality_id)
+    def buy(personality_id, price = nil)
+        if(price == nil || price <= 0) 
+            price = 200
+        end
         if(ActiveRecord::Base.connection.select_one("SELECT COUNT(*) count FROM personalities_users WHERE user_id = #{@user.id} AND personality_id = #{personality_id}")['count'] === 0 and @user.gems >= 200)
```

Comment:
```
 if(ActiveRecord::Base.connection.select_one("SELECT COUNT(*) count FROM personalities_users WHERE user_id = #{@user.id} AND personality_id = #{personality_id}")['count'] === 0 and @user.gems >= 200)
```
I'm not sure `@user.gems >= 200` , is it correct? I gess this is  `@user.gems >= price`